modules/v1/workspaces/models/Folder.php

```diff
@@ -0,0 +1,219 @@
+            'body' => Yii::t('app', 'Body'),
+            'file_path' => Yii::t('app', 'File Path'),
+            'mime' => Yii::t('app', 'Mime'),
+            'content' => Yii::t('app', 'Content'),
+            'size' => Yii::t('app', 'Size'),
+            'lft' => Yii::t('app', 'Lft'),
+            'rgt' => Yii::t('app', 'Rgt'),
+            'depth' => Yii::t('app', 'Depth'),
+            'tree' => Yii::t('app', 'Tree'),
+            'created_at' => Yii::t('app', 'Created At'),
+            'updated_at' => Yii::t('app', 'Updated At'),
+            'created_by' => Yii::t('app', 'Created By'),
+            'updated_by' => Yii::t('app', 'Updated By'),
+        ];
+    }
+
+    /**
+     * @return FolderQuery
+     */
+    public static function find()
+    {
+        return new FolderQuery(get_called_class());
+    }
+
+    /**
+     * Gets query for [[CreatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getCreatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'created_by']);
+    }
+
+    /**
+     * Gets query for [[Parent]].
+     *
+     * @return ActiveQuery
+     */
+    public function getParent()
+    {
+        return $this->hasOne(Folder::class, ['id' => 'parent_id']);
+    }
+
+    /**
+     * Gets query for [[Folders]].
+     *
+     * @return ActiveQuery
+     */
+    public function getChildren()
+    {
+        return $this->hasMany(Folder::class, ['parent_id' => 'id']);
+    }
+
+    /**
+     * Gets query for [[UpdatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getUpdatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'updated_by']);
+    }
+
+    /**
+     * Gets query for [[Workspace]].
+     *
+     * @return ActiveQuery
+     */
+    public function getTimelinePost()
+    {
+        return $this->hasOne(TimelinePost::class, ['id' => 'timeline_post_id']);
+    }
+
+    /**
+     * Gets query for [[Workspace]].
+     *
+     * @return ActiveQuery
+     */
+    public function getWorkspace()
+    {
+        return $this->hasOne(Workspace::class, ['id' => 'workspace_id']);
+    }
+
+    /**
+     * Upload file
+     *
+     * @param UploadedFile $file
+     * @return bool
+     * @throws Exception
+     */
+    public function uploadFile(UploadedFile $file)
+    {
+        $path = '/files';
+        $fullPath = Yii::getAlias('@storage' . $path);
+
+        if (!file_exists($fullPath) && !FileHelper::createDirectory($fullPath)) {
+            throw new ValidationException(Yii::t('app', 'Unable to upload file'));
+        }
+
+        $this->name = $file->name;
+        $this->mime = $file->type;
+        $this->size = $file->size;
+        $this->file_path = $path . '/' . Yii::$app->security->generateRandomString() . '.' . $file->extension;
+
+        if (!$file->saveAs(Yii::getAlias('@storage/' . $this->file_path))) {
+            throw new ValidationException(Yii::t('app', "File was NOT uploaded at '$this->file_path'"));
```

Comment:
@skavt We should not return user the file path in error message. Just return `Could not save file`